src/Workspaces/VisualBasic/Portable/Formatting/Engine/Trivia/TriviaDataFactory.Analyzer.vb

```diff
@@ -71,19 +72,25 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Formatting
                 result.Tab = 0
             End Sub
 
-            Private Shared Sub AnalyzeLineBreak(trivia As SyntaxTrivia, ByRef result As AnalysisResult)
-                ' if there was any space before line break, then we have trailing spaces
-                If result.Space > 0 OrElse result.Tab > 0 Then
-                    result.HasTrailingSpace = True
+            Private Shared Sub AnalyzeLineBreak(previousTrivia As SyntaxTrivia, trivia As SyntaxTrivia, ByRef result As AnalysisResult)
+                ' if there was any space immediately before line break, then we have trailing spaces
+                Dim text = previousTrivia.ToString()
+                If previousTrivia.Kind = SyntaxKind.WhitespaceTrivia Then
+                    For i As Integer = 0 To previousTrivia.Width - 1
+                        If text(i) = " "c OrElse text(i) = vbTab Then
+                            result.HasTrailingSpace = True
+                            result.HasTabAfterSpace = False
+                            result.Space = 0
+                            result.Tab = 0
+                            result.TreatAsElastic = result.TreatAsElastic Or trivia.IsElastic()
+                            Exit For
+                        End If
```

Comment:
i legitimately do not understand what is going on here.  We have trivia that we know is whitespace (it's WhitespaceTrivia).  So we start walking over it's width, and we get each character.  And if that character is whitespace... then we set all these values.  But don't know we know that it's whitespace already?

Reply:
The original code checked 2 variables result.space and result.tab which look at all the trivia between 2 tokens, I just copied the code that set those variables a bad idea. I only want to look at the previous trivia and if its whitespace with width > 0 then I want to set all these variable. I have reworked the code to remove the loop and just look at previousTrivia.Widtth.